lib/src/presentation/bloc/movie_controller.dart

```diff
@@ -50,4 +52,24 @@ class MovieController extends IMovieController {
 
   @override
   Stream<MovieState> get moviesStream => _moviesController.stream;
+
+  Future<void> searchMovie(
+    String searchQuery,
+    String category,
+  ) async {
+    final List<Movie> searchResult = [];
+    movies ??= await getMoviesUseCase.getMovieList(category);
+    final totalMovies = movies!.data!;
+    _moviesController.sink.add(MovieState.loading());
+    if (searchQuery.isEmpty) {
+      _moviesController.sink.add(MovieState.success(movieList: totalMovies));
```

Comment:
you can use an streabuilder that to avoid the need to call _moviesController.sink.add() manually.